### PR TITLE
Add unit tests for xtext-gradle-plugin

### PR DESCRIPTION
**Pull request by franzbecker**
I wrote a couple of unit tests for the plugins added by `xtext-gradle-plugin`.
They verify some basic behavior, e.g. that they can be applied without exception and so on.

What do you think?

### CODE REVIEW

File: xtext-gradle-plugin/src/test/java/org/xtext/gradle/AbstractPluginTest.xtend [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	def void noExceptionOnApply() {
+		// when
+		project.apply(pluginClass)
+	}
+
+	/**
+	 * No exception is thrown if plugin is applied twice to a project.
+	 */
+	@Test
+	def void applyIsIdempotent() {
```
Comment (oehme) [inline xtext-gradle-plugin/src/test/java/org/xtext/gradle/AbstractPluginTest.xtend]: Idempotency is more than not throwing exceptions. Maybe call this "canBeAppliedTwiceWithoutErrors"?

File: xtext-gradle-plugin/src/test/java/org/xtext/gradle/AbstractPluginTest.xtend [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	protected def void apply(Project project, Class<? extends Plugin<?>> pluginClass) {
+		project.apply[plugin(pluginClass)]
+	}
+	
+	protected def void task(Project project, Class<? extends Task> taskClass, String taskName) {
+		project.task(#{'type' -> taskClass}, taskName)	
+	}
+
+	/**
+	 * No exception is thrown if plugin is applied to a project.
```
Comment (oehme) [inline xtext-gradle-plugin/src/test/java/org/xtext/gradle/AbstractPluginTest.xtend]: Can you change the method names so no extra comment is needed? e.g. shouldNotThrowExceptionsOnApply

File: xtext-gradle-plugin/src/test/java/org/xtext/gradle/AbstractPluginTest.xtend [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+		return ProjectBuilder.builder.withName(name).withProjectDir(new File(projectDir, name)).withParent(
+			parentProject).build
+	}
+
+	protected def void apply(Project project, Class<? extends Plugin<?>> pluginClass) {
+		project.apply[plugin(pluginClass)]
+	}
+	
+	protected def void task(Project project, Class<? extends Task> taskClass, String taskName) {
+		project.task(#{'type' -> taskClass}, taskName)
```
Comment (oehme) [inline xtext-gradle-plugin/src/test/java/org/xtext/gradle/AbstractPluginTest.xtend]: This doesn't seem to be used anywhere and I don't know why I would create a Task from the test code?

Comment (franzbecker) [inline xtext-gradle-plugin/src/test/java/org/xtext/gradle/AbstractPluginTest.xtend]: You are right, I intended to use this when testing tasks but I didn't so far. In any case this shouldn't be in this class so I removed it.

**Comment by oehme** [comment]
Yes, some basic sanity checking is a good idea. Unfortunately, unit-testing the plugins is pretty limited, since not even the afterEvaluate-hooks are run.

There is some logic worthy of testing outside of the end-to-end tests. Some of it needs to be extracted from tasks first:

- writing Eclipse settings files 
- finding the correct intellij distribution url, given a version
- finding intellij plugins. given a repository and a plugin id+version
- does the GradleWorkspaceConfig return the expected values

**Comment by oehme** [comment]
Thanks again =)

**Pull request merged**